external/PATCH-01-STRATUM

```diff
@@ -2975,11 +2975,22 @@ index 9f4bb3a9..f07e4843 100644
 +       goto internal_error;
 +    }
  
++    auto keys_begin=keys.begin();
++    auto data_begin=data.begin();
++    std::advance(keys_begin,actual);
++    std::advance(data_begin,actual);
      table_keys->insert(table_keys->end(), std::make_move_iterator(keys.begin()),
-                        std::make_move_iterator(keys.end()));
-@@ -546,96 +599,139 @@ template <typename T>
+-                       std::make_move_iterator(keys.end()));
++                        std::make_move_iterator(keys_begin));
+     table_datums->insert(table_datums->end(),
+                          std::make_move_iterator(data.begin()),
+-                         std::make_move_iterator(data.end()));
++                         std::make_move_iterator(data_begin));
++
+   }
+ 
    CHECK(table_keys->size() == table_datums->size());
-   CHECK(table_keys->size() == entries);
+-  CHECK(table_keys->size() == entries);
```

Comment:
Why are we removing this check?

Reply:
table_keys size is no more equal to entries